scripts/utils/generate_strings.py

```diff
@@ -36,7 +36,31 @@ def construct_mapping(self, node, deep=False):
         return super().construct_mapping(node, deep)
 
 
-def parseTranslationStrings(yamlfile):
+def parseXLIFFTranslationStrings(xliff_file):
+    if not os.path.isfile(xliff_file):
+        exit(f"Unable to find {xliff_file}")
+
+    strings = {}
+
+    ns = {'x':'urn:oasis:names:tc:xliff:document:1.2'}
+    tree = etree.parse(xliff_file)
+    root = tree.getroot()
+
+    for node in root.xpath('//x:trans-unit', namespaces=ns):
+        id = node.get('id')
+        cpp_id = pascalize(id.replace('.', '_'))
```

Comment:
Oh :eyes: I am pretty sure this is not fine. Good call! Will fix.

Reply:
Well, hum. I wonder actually. Because this is building just fine... VSCode color codes the special character variant name differently, but maybe VSCode is just confused. I will check C++ specs before I fix this.

Reply:
The spec seems to leave it up to the compiler... I will fix it :/

Reply:
This is being extracted from the XLIFF id attribute. I am assuming in there it is not an issue, but let me know.

Reply:
Yeah, exactly. My question is if it an issue that the XLIFF file has ids with unicode characters i.e. should I fix that in the l10n repo as well.

Reply:
No, that's not an issue for our side.